src/metabase/metabot/util.clj

```diff
@@ -41,57 +42,76 @@
+                           (get-in (qp dataset_query nil nil) [:query :fields]))
+        field-ref->alias (reduce
+                           (fn [acc [_f _id-or-name m :as field-ref]]
+                             (if-let [alias (::add/desired-alias m)]
+                               (assoc acc (mbql.u/remove-namespaced-options field-ref) alias)
+                               acc))
+                           {}
+                           fields)]
+    (update model :result_metadata
+            (fn [result_metadata]
+              (map
+                (fn [{:keys [field_ref] :as rsmd}]
+                  (assoc rsmd :qp_column_name (field-ref->alias field_ref)))
+                result_metadata)))))
 
-(defn inner-query
+(defn- add-inner-query
   "Produce a SELECT * over the parameterized model with columns aliased to normalized display names.
+  Add this result to the input model along with the generated column aliases.
   This can be used in a CTE such that an outer query can be called on this query."
-  [{:keys [id] :as model}]
-  (let [column-aliases (->> (aliases model)
-                            (map (partial apply format "\"%s\" AS %s"))
-                            (str/join ","))]
-    (mdb.query/format-sql
-     (format "SELECT %s FROM {{#%s}} AS INNER_QUERY" column-aliases id))))
+  [{:keys [id result_metadata] :as model}]
+  (let [column-aliases (or
+                         (some->> result_metadata
+                                  (map (comp
+                                         (fn [[column_name column_alias]]
+                                           (cond
+                                             (and column_name column_alias) (format "\"%s\" AS %s" column_name column_alias)
+                                             column_alias column_alias
+                                             :else nil))
+                                         (juxt :qp_column_name :sql_name)))
+                                  (filter identity)
+                                  seq
+                                  (str/join ", "))
+                         "*")]
+    (assoc model
+      :column_aliases column-aliases
+      :inner_query
+      (mdb.query/format-sql
+        (format "SELECT %s FROM {{#%s}} AS INNER_QUERY" column-aliases id)))))
 
-(defn denormalize-field
+(defn- denormalize-field
   "Create a 'denormalized' version of the field which is optimized for querying
   and prompt engineering. Add in enumerated values (if a low-cardinality field),
-  a sql-friendly name, and remove fields unused in prompt engineering."
-  [{:keys [display_name id base_type] :as field}]
-  (let [field-vals (when (not= :type/Boolean base_type)
+  and remove fields unused in prompt engineering."
+  [{:keys [id base_type] :as field}]
+  (let [field-vals (when
+                     (and
+                       (not= :type/Boolean base_type)
+                       (< 0
```

Comment:
using `<=` could let you remove the `inc` here

Reply:
I'd just have to make the 0 a -1. 🤷